base-system/tools/installer/change-password.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+#	change-password.sh
+#	Script to change the password stored in the core files of a huronOS distribution
+#
+#	Copyright (C) 2023, huronOS Project:
+#		<http://huronos.org>
+#
+#	Licensed under the GNU GPL Version 2
+#		<http://www.gnu.org/licenses/gpl-2.0.html>
+#
+#	Authors:
+#		Daniel Cerna <dcerna@huronos.org>
+
+
+# Tests if a password was supplied
+if [ -z "$1" ]
+  then
+  	echo "Usage: change-password.sh your-new-password"
+  	exit 1
+fi
+
+## Test if the script is started by root user. If not, exit
+if [ "0$UID" -ne 0 ]; then
+	echo "Only root can run $(basename "$0")"; exit 1
+fi
+
+# check for mksquashfs with xz compression
+if [ "$(mksquashfs 2>&1 | grep "Xdict-size")" = "" ]; then
+   echo "mksquashfs not found or doesn't support -comp xz, aborting, no changes made"
+   echo "you may consider installing squashfs-tools package"
+   exit 1
+fi
+
+
+MAIN_PATH="$(pwd)"
+if [ -n "$2" ]; then
+  MAIN_PATH="$2"
+fi
+FILE_NAME="05-password.hsl"
+MAIN_FILE="$MAIN_PATH/huronOS/base/$FILE_NAME"
+
+LAB_DIR="/tmp/lab-$$"
+LAB_FILE="$LAB_DIR/$FILE_NAME"
+LAB_UNSQUASHED="$LAB_DIR/squashfs-root"
+SHADOW_FILE="$LAB_UNSQUASHED/etc/shadow"
+NEW_PASSWORD=$1
+SALT="\$6\$saltsalt\$"
+# Prepares the lab
+mkdir "$LAB_DIR"
+cd "$LAB_DIR" || exit 1
+cp "$MAIN_FILE" "$LAB_FILE"
+# Extracts the files
+unsquashfs "$MAIN_FILE"
+# Generate the encrypted password
+ENCRYPTED_PASSWORD=$(perl -e "print crypt('$NEW_PASSWORD', '$SALT')")
+echo "New password $NEW_PASSWORD"
+# Replace the password in the shadow file
+sudo sed -i "s|^\(root:\)\(\$[^:]*:\)|\1$ENCRYPTED_PASSWORD:|" "$SHADOW_FILE"
+# Ensure root is the owner of the shadow file
+sudo chown root "$SHADOW_FILE"
+# Ensure propper permissions on the shadow file
+sudo chmod 640 "$SHADOW_FILE"
+# Packages again the core files
+mksquashfs "$LAB_UNSQUASHED" "$LAB_FILE" -comp xz -b 1024K -always-use-fragments -noappend
```

Comment:
It would be nice to investigate how often do popular linux distros include the *squash fs* kernel module in their systems. Even if the module is now a default module, distros can configure it to change that. 
It would also be good to check that several distros have the `squashfs-tools` package available on their repos as we would be depending on that utility for using this tool.

Reply:
I agree, on the things to check testing the installer on different distros this is one of the things I'll keep an eye on.